odoo/addons/base/ir/ir_mail_server.py

```diff
@@ -367,6 +367,7 @@ def build_email(self, email_from, email_to, subject, body, email_cc=None, email_
                 # so we fix it by using RFC2047 encoding for the filename instead.
                 part.set_param('name', filename_rfc2047)
                 part.add_header('Content-Disposition', 'attachment', filename=filename_rfc2047)
+                part.add_header('Content-ID', '<%s>' % fname)
```

Comment:
CIDs must be unique, and compatible with headers. Using the filename is not unique (unless you can prove it is in this case) and could break when it contains non-ascii (see why we compute filename_rfc2047 above).
We need to generate a unique ID, perhaps using the SHA1 hex of the file contents could serve this purpose.